# Termux/Bionic port: src/jlapi.c
#
# rr detach/teleport is Linux-glibc only.
--- a/src/jlapi.c
+++ b/src/jlapi.c
@@ -1070,7 +1070,7 @@
 // Actual definition in `ast.c`
 void jl_lisp_prompt(void);
 
-#ifdef _OS_LINUX_
+#if defined(_OS_LINUX_) && !defined(_OS_ANDROID_)
 static void rr_detach_teleport(void) {
 #define RR_CALL_BASE 1000
 #define SYS_rrcall_detach_teleport (RR_CALL_BASE + 9)
@@ -1116,7 +1116,7 @@
     // the execution where we actually need to exclude rr (e.g. because we're
     // testing for the absence of a memory-model-dependent bug).
     if (jl_options.rr_detach && jl_running_under_rr(0)) {
-#ifdef _OS_LINUX_
+#if defined(_OS_LINUX_) && !defined(_OS_ANDROID_)
         rr_detach_teleport();
         execv("/proc/self/exe", argv);
 #endif
